testsuite/tests/kuadrant/identical_hostname/test_auth_on_gw_and_route.py

```diff
@@ -0,0 +1,63 @@
+"""
+Tests behavior of using one HTTPRoute declaring the same hostname as parent Gateway related to AuthPolicy.
+https://github.com/Kuadrant/kuadrant-operator/blob/main/doc/auth.md#limitation-multiple-network-resources-with-identical-hostnames
+(second topology mentioned there)
+"""
+
+import time
+
+import pytest
+
+from testsuite.policy.authorization.auth_policy import AuthPolicy
+
+pytestmark = [pytest.mark.kuadrant_only]
+
+
+@pytest.fixture(scope="class", autouse=True)
+def authorization2(request, kuadrant, gateway, authorization_name2, openshift, label):
+    """2nd Authorization object (In case of Kuadrant AuthPolicy)"""
+    if kuadrant:
+        auth_policy = AuthPolicy.create_instance(openshift, authorization_name2, gateway, labels={"testRun": label})
+        auth_policy.authorization.add_opa_policy("rego", "allow = false")
+        request.addfinalizer(auth_policy.delete)
+        auth_policy.commit()
+        auth_policy.wait_for_ready()
+        return auth_policy
+    return None
+
+
+def test_gateway_authpolicy_ignored(client, authorization):
+    """
+    Tests that Gateway-attached AuthPolicy is ignored on 'route2' if both 'route' and 'route2' declare
+    identical hostname and there is another AuthPolicy already successfully enforced on 'route'.
+    Setup:
+        - Two HTTPRoutes declaring identical hostnames but different paths ('/' and '/anything/')
+        - Empty AuthPolicy enforced on the '/' HTTPRoute
+        - 'deny-all' AuthPolicy created after Empty AuthPolicy enforced on the Gateway
+    Test:
+        - Send a request via 'route' and assert that response status code is 200
+        - Send a request via 'route2' and assert that response status code is 200
+        - Delete the empty AuthPolicy created via fixture
+        - Send a request via both routes
+        - Assert that both response status codes are 403 (Forbidden)
+    """
+
+    # Access via 'route' is allowed due to Empty AuthPolicy
+    response = client.get("/get")
+    assert response.status_code == 200
+
+    # Despite 'deny-all' Gateway AuthPolicy reporting being successfully enforced
+    # it is still allowed to access the resources via 'route2'
+    response = client.get("/anything/get")
+    assert response.status_code == 200
+
+    # Deletion of Empty AuthPolicy should make the 'deny-all' Gateway AuthPolicy effectively enforced.
+    # It might take some time hence the 15s sleep.
+    authorization.delete()
+    time.sleep(15)
```

Comment:
This will be flaky, we should find better solution that this. Cannot you check the GW AuthPolicy if it is still partially or fully enforced?

Reply:
No, not really. The GW AuthPolicy status is fully enforced even before the deletion so it can't be used. I share your concerns about stability though, I was just not able to come up with better solution.

Reply:
How about using retry for the next request?

Reply:
+1 I like that, sounds like a better approach to me. Will refactor as suggested

Reply:
What is the best way to do this? I added `client.add_retry_code(200)` but not 100% sure if it does the right thing. Alternatively I can use some retry_strategy as described e.g. here:
https://scrapeops.io/python-web-scraping-playbook/python-httpx-retry-failed-requests/

Reply:
Updated using a new retry client